specification/containerservice/Fleet.Management/fleetnamespace.tsp

```diff
@@ -120,28 +121,241 @@ union EgressPolicy {
+
+  @doc("PickFixed selects a fixed set of clusters")
+  PickFixed: "PickFixed",
+
+  string,
+}
+
+@added(Versions.v2025_03_02_naar)
+@doc("ClusterResourcePlacementAffinity is a group of cluster affinity scheduling rules.")
+model ClusterResourcePlacementAffinity {
+  @doc("ClusterAffinity contains cluster affinity scheduling rules for the selected resources.")
+  clusterAffinity?: ClusterResourcePlacementClusterAffinity;
+}
+
+@added(Versions.v2025_03_02_naar)
+@doc("ClusterAffinity contains cluster affinity scheduling rules for the selected resources.")
+model ClusterResourcePlacementClusterAffinity {
+  @doc("If any affinity requirements specified by this field are not met at scheduling time, the resource will not be scheduled onto the cluster.")
+  requiredDuringSchedulingIgnoredDuringExecution?: CrpClusterSelector;
+}
+@added(Versions.v2025_03_02_naar)
+@doc("CrpClusterSelector allows specifying cluster selector terms.")
+model CrpClusterSelector {
+  @doc("ClusterSelectorTerms is a list of cluster selector terms. The terms are `ORed`.")
+  @OpenAPI.extension("x-ms-identifiers", [])
+  clusterSelectorTerms: CrpClusterSelectorTerm[];
+}
+
+@added(Versions.v2025_03_02_naar)
+@doc("CrpClusterSelectorTerm allows specifying what clusters in the fleet are in scope. If you specify both label and property selectors in the same term, the results are AND'd.")
+model CrpClusterSelectorTerm {
+  @doc("LabelSelector is a label query over all the joined member clusters. Clusters matching the query are selected.")
+  labelSelector?: MetaV1ClusterLabelSelector;
+
+  @doc("PropertySelector is a property query over all joined member clusters. Clusters matching the query are selected.")
+  propertySelector?: CrpClusterPropertySelector;
+}
+
+@added(Versions.v2025_03_02_naar)
+@doc("A label selector is a label query over a set of resources. The result of matchLabels and matchExpressions are ANDed. An empty label selector matches all objects. A null label selector matches no objects.")
+model MetaV1ClusterLabelSelector {
   #suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" "Records is the most semantically correct choice for this use-case"
-  @doc("The annotations for the fleet managed namespace.")
-  annotations?: Record<string>;
+  @doc("matchLabels is a map of {key,value} pairs.")
+  matchLabels?: Record<string>;
 
-  @doc("The default resource quota for the fleet managed namespace.")
-  defaultResourceQuota?: ResourceQuota;
+  @doc("matchExpressions is a list of label selector requirements. The requirements are ANDed.")
+  @OpenAPI.extension("x-ms-identifiers", [])
+  matchExpressions?: MetaV1LabelSelectorRequirement[];
+}
 
-  @doc("The default network policy for the fleet managed namespace.")
-  defaultNetworkPolicy?: NetworkPolicy;
+@added(Versions.v2025_03_02_naar)
+@doc("A label selector requirement is a selector that contains values, a key, and an operator that relates the key and values.")
+model MetaV1LabelSelectorRequirement {
```

Comment:
is there any reason not just call it LabelSelectorRequirement?

Reply:
Yes, the intention is to fix all of these up.

> Let me rephase my question, how many hours would it take me to write this PR without any bug (as it's a perfect copy paste of CRP API)?

The intention is to get this homogeneous enough such that the process of translating future API updates into this typespec could in principle be subsequently automated.

Reply:
This is now fixed.